src/components/fileviewer.js

```diff
@@ -33,29 +33,37 @@ export default class FileViewerContainer extends Component {
       });
     });
 
-    FetchAPI.getFileRevisionCoverage(revision, path,
-      (data) => {
-        console.log(data);
-        this.setState(() => ({ coverage: data }));
-        // TODO remove these log lines
-        console.log(this.state.coverage.data[0][0].source.file.covered);
-        console.log(this.state.coverage.data[0][0].source.file.uncovered);
-        console.log(this.state.coverage.data[0][0].source.file.percentage_covered);
-      }
-    )
+    FetchAPI.getFileRevisionCoverage(revision, path)
+      .then(response => {
+        if (response.status !== 200) {
```

Comment:
Put this error handling in the `getFileRevisionCoverage` function.

Reply:
I see what Arman doing is to keep the file_data.js simple and clean, and put the error handling, catch codes in the viewer function. How about create a separate js file to holds the API fetching calls for ucosp project?

Reply:
My philosophy is to make a facade for every library function you use.  This will isolate you from inevitable anomalies those libraries have in the corner cases.  This facade can be further enhanced it to deal with your application-specific usage patterns.

You will now see that `FetchAPI.getRawFile` and `FetchAPI.getFileRevisionCoverage` have common code.  Put this common code in its own function (maybe called `FetchAPI.post`), which will be the beginning of the "facade" I talk about.  

When you have done that, the `FetchAPI.getRawFile` and `FetchAPI.getFileRevisionCoverage` should be single-line functions that are used in only one place in your code:  Inline them.